plugins/techdocs-backend/src/cache/cacheMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Router } from 'express';
+import router from 'express-promise-router';
+import { Logger } from 'winston';
+import { TechDocsCache } from './TechDocsCache';
+
+type CacheMiddlewareOptions = {
+  cache: TechDocsCache;
+  logger: Logger;
+};
+
+type ErrorCallback = (err?: Error) => void;
+
+export const createCacheMiddleware = ({
+  cache,
+}: CacheMiddlewareOptions): Router => {
+  const cacheMiddleware = router();
+
+  // Middleware that, through socket monkey patching, captures responses as
+  // they're sent over /static/docs/* and caches them. Subsequent requests are
+  // loaded from cache. Cache key is the object's path (after `/static/docs/`).
+  cacheMiddleware.use(async (req, res, next) => {
+    const socket = res.socket;
+    const isCacheable = req.path.includes('/static/docs/');
```

Comment:
Would it be possible to use `startsWith`? Should be unlikely but prevents from unrelated resources to be marked cacheable

Reply:
I had assumed that `req.path` would include the full `path`-part of the request URL, but it turns out that in the context of express middleware, the mount point is removed; so `startsWith` is indeed the safer option!  Will update.

From express docs: 

> _When called from a middleware, the mount point is not included in req.path. See app.use() for more details._

Reply:
^ It appears supertest doesn't actually behave this way, but the real app does, at least.  So I tweaked the tests to be conformant.